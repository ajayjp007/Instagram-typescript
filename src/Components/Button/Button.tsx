import './Button.css';
import React, { useState } from 'react';

const Button: React.FC<{ content: string; id: string }> = ({ content, id }) => {
  const [sendFriendRequest, setSendFriendRequest] = useState<boolean>(false);
  const addFriendHandler = (event: any) => {
    setSendFriendRequest(!sendFriendRequest);
    const myHeaders = new Headers();
    myHeaders.append('Content-Type', 'application/json');

    const raw = JSON.stringify({
      id: event.target.id,
      username: localStorage.getItem('userName'),
    });

    const requestOptions: any = {
      method: 'POST',
      headers: myHeaders,
      body: raw,
      redirect: 'follow',
    };

    fetch('http://localhost:5000/api/users/add-new-friend', requestOptions)
      .then((response) => response.text())
      .then((result) => result)
      .catch((error) => error);
  };
  return (
    <button
      type="button"
      id={id}
      className={sendFriendRequest ? 'follow-btn friendReqSent' : 'follow-btn'}
      onClick={addFriendHandler}
    >
      {!sendFriendRequest ? content : 'Requested'}
    </button>
  );
};

export default Button;
