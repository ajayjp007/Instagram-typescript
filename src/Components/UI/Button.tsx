import "./Button.css";
import React, { useState } from "react";

const Button: React.FC<{ content: string }> = (props) => {
  const [sendFriendRequest, setSendFriendRequest] = useState<boolean>(false);
  const addFriendHandler = () => {
    setSendFriendRequest(!sendFriendRequest);
  };
  return (
    <button
      type="button"
      className={sendFriendRequest ? "follow-btn friendReqSent" : "follow-btn"}
      onClick={addFriendHandler}
    >
      {!sendFriendRequest ? props.content : "Requested"}
    </button>
  );
};

export default Button;
