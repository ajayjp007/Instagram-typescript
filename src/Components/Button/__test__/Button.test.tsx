import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import Button from '../Button';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

describe('Button component test', () => {
  afterEach(cleanup);
  test('should render button component', async () => {
    render(<Button content="follow" id="some-random-id" />);
    const buttonElement = screen.getByTestId('Button-1');
    fireEvent.click(
      buttonElement,
      (global.fetch = jest.fn().mockImplementationOnce(() =>
        Promise.resolve({
          status: 400,
          json: () =>
            Promise.resolve({
              success: false,
              error: 'Something bad happened',
            }),
        }),
      )),
    );
    expect(buttonElement).toHaveTextContent('Requested');
  });

  test('should render button text properly', () => {
    const { getByTestId } = render(
      <Button content="follow" id="some-random-id" />,
    );
    expect(getByTestId('Button-1')).toHaveTextContent('follow');
  });

  test('matches snapshot', () => {
    const ButtonSnapshot = renderer
      .create(<Button content="follow" id="some-random-id" />)
      .toJSON();
    expect(ButtonSnapshot).toMatchSnapshot();
  });
});
