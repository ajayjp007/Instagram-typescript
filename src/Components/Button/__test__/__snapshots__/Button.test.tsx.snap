// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test matches snapshot 1`] = `
<button
  className="follow-btn"
  data-testid="Button-1"
  id="some-random-id"
  onClick={[Function]}
  type="button"
>
  follow
</button>
`;
